### PR TITLE
fix(input): textarea style bug

### LINKED ISSUE #41125: Input 组件 TextArea添加删除按钮， 文字太多会遮挡住删除按钮
### Reproduction link

[https://ant.design/components/input-cn#inputtextarea](https://ant.design/components/input-cn#inputtextarea)

image

### Steps to reproduce

官网demo即可复现

### What is expected?

文字不遮挡删除按钮

### What is actually happening?

文字遮挡了

| Environment | Info |
| --- | --- |
| antd | 5.3.0 |
| React | React17/18 |
| System | Mac |
| Browser | chrom |

### PR DESCRIPTION
**Pull request by Ali-ovo**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #41125

### 💡 Background and solution

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  fix style bug         |
| 🇨🇳 Chinese |  修复文字覆盖关闭按钮的问题         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by Ali-ovo** [comment]
close https://github.com/ant-design/ant-design/issues/41125

关联 https://github.com/react-component/textarea/pull/35

**Comment by Ali-ovo** [comment]
@MadCcc `基于feature分支` 是这样没错吧？

**Review by li-jia-nan — APPROVED** [review]

File: components/input/style/index.ts [hunk 1]
```diff
@@ -890,6 +890,13 @@ const genTextAreaStyle: GenerateStyle<InputToken> = (token) => {
         },
       },
 
+      '&-allow-clear': {
+        [`> ${componentCls}`]: {
+          paddingInlineEnd: paddingLG,
+          overflow: 'overlay',
```
[the remark this answers is not in the document]
Comment (Ali-ovo) [reply components/input/style/index.ts:896]: 如果不使用这个属性就会导致右边距padding过大 
image

有什么其他更好的解决方案么

[the remark this answers is not in the document]
Comment (MadCcc) [reply components/input/style/index.ts:896]: 可以考虑把 clear icon 改成 static 布局，占宽度

[the remark this answers is not in the document]
Comment (Ali-ovo) [reply components/input/style/index.ts]: 目前这个 icon 是通过absloute定位过去的 有点麻烦 
我先把 `overlay` 去掉吧

[the remark this answers is not in the document]
Comment (Ali-ovo) [reply components/input/style/index.ts]: done

**Review by afc163 — APPROVED** [review]

**Pull request merged**